### DISCUSSION TITLE
Feedback on "Squash and Merge" in Merge Queue (Beta)

### DISCUSSION THREAD

**Discussion by mhoeger** [community/community#14818] [Pull Requests]
Super excited about Merge Queue!! Loving it already. Hoping this is an appropriate place for feedback on the feature.

Our team uses the `Squash and Merge` flow, but we also rely on PR titles to become the commit message (conventional commits). Unfortunately, the `Squash and Merge` flow ignores PR title and takes first commit message (I believe).

Would love for there to be some opportunity to edit the commit message (same as today for normal squash and merge) before popping it in the queue!

**Comment by discoivanksk** [comment]
hello dear GitHub team, we are so excited to use merge queue, it is really amazing and wonderful feature, thank you very much for your hardwork to bring this feature to us, however we find that ability to have `squash and merge` might be very beneficial, since less bytes will be saved in git log, meaning it will be more `eco-friendly and follow green tech approach (less data transferred and stored=less processor time and energy used, very important in energy shortage times)`, compare to multiple records storage, is it possible to prioritize it? Kind regards, Ivan
PS thank you for you hard work on merge queue functionality

**Comment by csantos1113** [comment]
> Unfortunately, the Squash and Merge flow ignores PR title and takes first commit message (I believe).

We were experiencing the same behavior, but apparently it changed recently?
Now it uses the title of the PR, and all commits are part of the commit's body.

I wonder if there is a way we can see the updates GitHub makes to the `merge queue (beta)` feature so we don't hit these changes randomly

**Reply by willsmythe** [reply]
There has been no intentional change in behavior here. The message for the squash commit created by merge queue should be the same as the message that would be presented if merging the same pull request without merge queue. And like with regular merging, the default commit message will change based on the number of commits in the pull request (see [docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#merge-message-for-a-squash-merge)). If you are not seeing this behavior, can you reach out to [support](https://support.github.com/request/pull-requests) and provide more details?

**Reply by inikolaev** [reply]
Would be great to have the same behaviour a with regular auto merges when it's possible to customise the title and the message. Now it only shows a green confirmation button when enqueueing PR into the merge queue

**Reply by zzJinux** [reply]
@willsmythe Hey, I created a ticket for it. I hope it prioritize the issue: https://support.github.com/ticket/personal/0/3588417

**Comment by Hona** [comment]
Any update on this? Its a pretty big showstopper for merge queues

**Comment by tadeuszkleszcz** [comment]
I have checked the behavior with the merge queue and squash commit features enabled.
There is different behavior for PRs with single and multiple commits.
- After merging PR with a single commit the new commit in the target branch has the same commit message as the commit in the source branch and the PR title is ignored.
- After merging PR with multiple commits the new commit message is taken from PR title.

Currently, there is no option to customize the squash commit message after enabling the merge queue.
When the merge queue is disabled even PR with a single commit can have a commit message customized in GitHub UI during merging.

**Reply by tadeuszkleszcz** [reply]
I have found out that after setting the default commit message when merging PR it works correctly for single commit branches - PR title is taken to squash the commit message in the new branch. More details here: https://github.blog/changelog/2022-08-23-new-options-for-controlling-the-default-commit-message-when-merging-a-pull-request/.
